ddtrace/internal/datadog/profiling/dd_wrapper/src/uploader_builder.cpp

```diff
@@ -0,0 +1,124 @@
+#include "uploader_builder.hpp"
+#include "libdatadog_helpers.hpp"
+#include "types.hpp"
+
+#include <algorithm>
+#include <mutex>
+#include <string_view>
+
+using namespace Datadog;
+
+UploaderBuilder&
+UploaderBuilder::set_env(std::string_view _env)
+{
+    std::lock_guard<std::mutex> lock(mtx);
+    if (!_env.empty())
+        env = _env;
+    return *this;
+}
+UploaderBuilder&
+UploaderBuilder::set_service(std::string_view _service)
+{
+    std::lock_guard<std::mutex> lock(mtx);
+    if (!_service.empty())
+        service = _service;
+    return *this;
+}
+UploaderBuilder&
+UploaderBuilder::set_version(std::string_view _version)
+{
+    std::lock_guard<std::mutex> lock(mtx);
+    if (!_version.empty())
+        version = _version;
+    return *this;
+}
+UploaderBuilder&
+UploaderBuilder::set_runtime(std::string_view _runtime)
+{
+    std::lock_guard<std::mutex> lock(mtx);
+    if (!_runtime.empty())
+        this->runtime = _runtime;
+    return *this;
+}
+UploaderBuilder&
+UploaderBuilder::set_runtime_version(std::string_view _runtime_version)
+{
+    std::lock_guard<std::mutex> lock(mtx);
+    if (!_runtime_version.empty())
+        runtime_version = _runtime_version;
+    return *this;
+}
+UploaderBuilder&
+UploaderBuilder::set_profiler_version(std::string_view _profiler_version)
+{
+    std::lock_guard<std::mutex> lock(mtx);
```

Comment:
are you going to repeatedly acquire and release the mutex if you have `.set_profiler_version().set_runtime_version()` etc?

Reply:
Indeed, and doing so would be rather pointless (it's a loser-takes-all situation no matter how you slice the variants here, since we are doing trivial assignment).

I removed these locks.

Reply:
Removed